src/system_identity.rs

```diff
@@ -1 +1,81 @@
+pub mod creation_response;
+pub mod generator;
+pub mod identity_creator;
 pub mod input_data;
+
+/// System identity information. Final output of the System Identity creation process.
+#[derive(Debug, Clone)]
+pub struct SystemIdentity {
+    pub name: String,
+    pub client_id: String,
+    pub identity_type: SystemIdentityType,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum SystemIdentityType {
+    L1 {
+        id: String,
+        client_secret: ClientSecret,
+    },
+    L2 {
+        pub_key: PublicKey,
```

Comment:
We won't get the public key back if we don't modify the request https://github.com/newrelic/newrelic-auth-rs/pull/25/files#diff-ac203ab36353afa8ffb81298ffcaebf3b013fb82cae874c86753fb4e1b754948R56.